src/shims/unix/sync.rs

```diff
@@ -176,10 +176,11 @@ fn mutex_create<'tcx>(
     let mutex = ecx.deref_pointer(mutex_ptr)?;
     let id = ecx.machine.sync.mutex_create();
     let data = PthreadMutex { id, kind };
-    ecx.lazy_sync_init(&mutex, mutex_init_offset(ecx)?, data)?;
+    ecx.lazy_sync_init(&mutex, mutex_init_offset(ecx)?, data.clone())?;
     interp_ok(data)
 }
 
+// TODO: check if the comment is correct?
 /// Returns the `MutexId` of the mutex stored at `mutex_op`.
 ///
 /// `mutex_get_id` will also check if the mutex has been moved since its first use and
```

Comment:
This comment looks a bit weird. ``mutex_get_data`` returns a ``PthreadMutex``, and why is there ``mutex_op`` and ``mutex_get_id``?

Reply:
Yeah that comment is just outdated. Make it like this:
```
/// Returns the mutex data stored at the address that `mutex_ptr` points to.
/// Will raise an error if the mutex has been move since its first use.
```